### PR TITLE
[RDY] Move hardcoded hospital mapgen to json

### LINKED ISSUE #18924: No toilets in a hospital
Well, that's kinda inconvenient ;)

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
# Description

Moved hospital mapgen to json.

It is not the same hospital as hardcoded one, but this one at least has toilets (fixes #18924) ;)

Please give your feedback for new layout!

# To-do

- [X] - Handle legacy hospital ids in savegames
- [X] - Removed hardcoded mapgen
- [X] - Add json mapgen
- [X] - Add `GROUP_HOSPITAL` monster group (same as `GROUP_ZOMBIE`, but without dogs)
- [X] - Add computers
- [x] - Add patient corpses
- [x] - Add loot
- [x] - Decided not to add basement this time

# Screenshots

default

default

default

### CODE REVIEW

**Comment by rm-code** [comment]
I really like it though in one of your screenshot there seems to be an error with the mapgen:

33524740-69cbd12c-d833-11e7-9e6b-d43d4c3c8db4

File: data/json/mapgen/hospital/hospital.json [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        "###########xxx//xxx#...##############....l#...#l....#``#....l#...#l....#",
+        "#l........x............x...........r#BB...+...+...BB#``#BB...+...+...BB#",
+        "#r.TBB7.../............/.....6.....r#BB...+...+...BB#``#BB...+...+...BB#",
+        "#r..BB..../............/.....*.....r#...cT#...#Tc...#``#...cT#...#Tc...#",
+        "#.........x............x...........r#######...#######``#######...#######",
+        "###########xxx//xxx#...##############...cT#...#Tc...#``#...cT#...#Tc...#",
+        "###########........#...x...........r#BB...+...+...BB#``#BB...+...+...BB#",
+        "###########r..BB...#.../......BB...r#BB...+...+...BB#``#BB...+...+...BB#",
+        "###########r.TBBT..#.../.....TBBT...#....l#...#l....#``#....l#...#l....#",
+        "###########.......l#...x...........l########...#######``#######...#######",
```
Comment (rm-code) [inline data/json/mapgen/hospital/hospital.json]: Looks like the lines in question are number 61 and 62 here.

Comment (ZhilkinSerg) [reply data/json/mapgen/hospital/hospital.json]: Thanks! I've fixed that.

**Comment by codemime** [comment]
This has a trivial merge conflict. Otherwise it looks good to go.

**Comment by codemime** [comment]
I suppose it's a fair trade-off for the missing toilets. Why don't you put the WIP tag then?

**Comment by ZhilkinSerg** [comment]
> I suppose it's a fair trade-off for the missing toilets. Why don't you put the WIP tag then?

I guess someone told me not to put both tags (`CR` and `WIP`). Anyway loot (and probably corpses too) was added.

**Pull request merged**